pyaerocom/io/test/test_readgridded.py

```diff
@@ -57,34 +58,120 @@ def test_ReadGridded_class_empty():
     assert r.data_dir == None
     from pyaerocom.io.aerocom_browser import AerocomBrowser
     assert isinstance(r.browser, AerocomBrowser)
-
-    failed = False
-    try:
+    with pytest.raises(AttributeError):
         r.years_avail
-    except AttributeError:
-        failed = True
-    assert failed
     assert r.vars_filename == []
 
+
+@testdata_unavail
+def test_ReadGridded_data_dir(reader_tm5):
+    assert reader_tm5.data_dir == path_tm5
+    assert reader_tm5._vars_2d == ['abs550aer', 'od550aer']
+    assert reader_tm5._vars_3d == []
+
+
+def test_ReadGridded_ts_types():
+    r = ReadGridded(data_dir=path_tm5)
+    assert sorted(r.ts_types) == ['daily', 'monthly']
+
+@testdata_unavail
+def test_ReadGridded_read_var(reader_tm5):
+    r = reader_tm5
+    data = r.read_var('od550aer')
+    assert isinstance(data, GriddedData)
+    ds = data.to_xarray().load()
+    mean = float(ds.mean(dim='time').mean(dim='lat').mean(dim='lon'))
+    npt.assert_almost_equal(mean, 0.09631748)
```

Comment:
Why converting to xarray and doing mean over individual dimensions? You could probably also do data.mean() directly (which will be area weighted, though, by default, but should suffice).

Reply:
Forgot that GriddedData has that functionality! The weird way of taking means is caused by some 4D data I worked with recently :)